### DISCUSSION TITLE
Some Errors in json resource collection and model laravel

### DISCUSSION THREAD

**Discussion by waelsjaber** [larastan/larastan#1189] [Q&A]
I use json resource collection laravel

Currently I am getting:

1. `Access to an undefined property App\Http\Resources\UserResource::$role => $role is pivot relationships`.
2. `Call to an undefined method App\Http\Resources\UserResource::relationLoaded()` .
3. `Call to an undefined static method App\Models\Address::findMultiWithPluck() => findMultiWithPluck() is function custom in MacrosServiceProvider` .

```php
class UserResource extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'role' => $this->role->name,
            'city_name' =>  $this->when($this->relationLoaded('city'),function(){
                    return $this->city->name
            }),
            'addresses' => Address::findMultiWithPluck($this->city->addresses()->pluck('id')->toArray()),
         ];
    }
}
```

is there a way to get this fixed?

**Comment by szepeviktor** [comment]
Hello @waelsjaber! 👋 
It is much easier to help you if you open one issue per problem with all details.

**Comment by canvural** [comment]
Hi,

You can add `@mixin User` to your class docblock to solve these.
